app/src/test/java/ch/hikemate/app/model/elevation/ElevationRepositoryCopernicusTest.kt

```diff
@@ -0,0 +1,312 @@
+            }
+
+        val onSuccessLatch = CountDownLatch(1)
+
+        elevationRepositoryCopernicus.getElevation(
+            twiceChunkSizedList,
+            { list ->
+              assertEquals(twiceChunkSizedList.map { it.lat }, list)
+              onSuccessLatch.countDown()
+            }) {
+              fail("Failed to fetch routes from Overpass API")
+            }
+
+        // Wait for the coroutine to finish
+        advanceUntilIdle()
+
+        assertEquals(0L, onSuccessLatch.count)
+        coVerify(exactly = 2) { call.enqueue(any()) }
+      }
+
+  @Test
+  fun worksWithSameRequestOnMultipleChunksMultipleTimesWithoutDelay() =
+      runTest(timeout = 5.seconds) {
+        val call: Call = mockk()
+        coEvery { client.newCall(any()) } returns call
+
+        coEvery { call.enqueue(any()) } answers
+            {
+              firstArg<Callback>().onResponse(call, twiceChunkSizedResponsePart1)
+            } andThenAnswer
+            {
+              firstArg<Callback>().onResponse(call, twiceChunkSizedResponsePart2)
+            }
+
+        val onSuccessLatch = CountDownLatch(2)
+
+        elevationRepositoryCopernicus.getElevation(
+            twiceChunkSizedList,
+            { list ->
+              assertEquals(twiceChunkSizedList.map { it.lat }, list)
+              onSuccessLatch.countDown()
+            }) {
+              fail("Failed to fetch routes from Overpass API")
+            }
+
+        elevationRepositoryCopernicus.getElevation(
+            twiceChunkSizedList,
+            { list ->
+              assertEquals(twiceChunkSizedList.map { it.lat }, list)
+              onSuccessLatch.countDown()
+            }) {
+              fail("Failed to fetch routes from Overpass API")
+            }
+
+        // Wait for the coroutine to finish
+        advanceUntilIdle()
+
+        assertEquals(0L, onSuccessLatch.count)
+        coVerify(exactly = 2) { call.enqueue(any()) }
+      }
+
+  @Test
+  fun worksWithSameRequestOnMultipleChunksMultipleTimesWithDelay() =
+      runTest(timeout = 5.seconds) {
+        val call: Call = mockk()
+        coEvery { client.newCall(any()) } returns call
+
+        coEvery { call.enqueue(any()) } answers
+            {
+              firstArg<Callback>().onResponse(call, twiceChunkSizedResponsePart1)
+            } andThenAnswer
+            {
+              firstArg<Callback>().onResponse(call, twiceChunkSizedResponsePart2)
+            }
+
+        val onSuccessLatch = CountDownLatch(2)
+
+        elevationRepositoryCopernicus.getElevation(
+            twiceChunkSizedList,
+            { list ->
+              assertEquals(twiceChunkSizedList.map { it.lat }, list)
+              onSuccessLatch.countDown()
+            }) {
+              fail("Failed to fetch routes from Overpass API")
+            }
+
+        // Wait for the coroutine to finish
+        while (onSuccessLatch.count > 1) {
+          advanceUntilIdle()
+        }
```

Comment:
Change: if the test breaks for some reason, and `onSuccess` is not called, this will go into an infinite loop. Add some kind of maximum iterations or timeout.

Reply:
Actually, there is already one! See in the definition of the test : `runTest(timeout = 5.seconds)`

Reply:
Oh right forgot about that one